thunderbird/processes/wps_generate_prsn.py

```diff
@@ -0,0 +1,231 @@
+        The DAP Standard specifies that a specific tag must be included in the
+        Content-Description header of every request. This tag is one of:
+            "dods-dds" | "dods-das" | "dods-data" | "dods-error"
+        So we can check if the header starts with `dods`.
+        Even then, some OpenDAP servers seem to not include the specified header...
+        So we need to let the netCDF4 library actually open the file.
+       """
+        try:
+            content_description = head(url, timeout=5).headers.get(
+                "Content-Description"
+            )
+        except (ConnectionError, MissingSchema, InvalidSchema):
+            return False
+
+        if content_description:
+            return content_description.lower().startswith("dods")
+        else:
+            try:
+                dataset = CFDataset(url)
+            except OSError:
+                return False
+            return dataset.disk_format in ("DAP2", "DAP4")
+
+    def get_filepaths(self, request):
+        filepaths = {}
+        var_list = ["pr", "tasmin", "tasmax"]
+        data_files = [
+            request.inputs["prec"][0],
+            request.inputs["tasmin"][0],
+            request.inputs["tasmax"][0],
+        ]
+        for var, path in zip(var_list, data_files):
+            if self.is_opendap_url(path.url):
+                filepaths[var] = path.url
+            else:
+                filepaths[var] = path.file
+        return filepaths
+   
+    def setup_logger(self, level):
+        formatter = logging.Formatter("%(asctime)s %(levelname)s: %(message)s",
+                                      "%Y-%m-%d %H:%M:%S")
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        logger.setLevel(level)
+
+    def collect_prsn_file(self):
+        return [file for file in os.listdir(self.workdir) if "prsn" in file][0]
+
+    def build_meta_link(self, prsn_file):
+        meta_link = MetaLink4(
+            "output", "Output of netCDF prsn file", workdir=self.workdir
+        )
+
+        # Create a MetaFile instance, which instantiates a ComplexOutput object.
+        meta_file = MetaFile(f"{prsn_file}", "Precipitation as snow", fmt=FORMATS.NETCDF)
+        meta_file.file = os.path.join(self.workdir, prsn_file)
+        meta_link.append(meta_file)
+
+        return meta_link.xml
+
+    def _handler(self, request, response):
+        response.update_status("Starting Process", 0)
+
+        (
+            chunk_size,
+            loglevel,
+            dry_run,
+            output_file
+        ) = self.collect_args(request)
+        output_file = None if output_file == "None" else output_file
+        filepaths = self.get_filepaths(request)
+        filepaths["pr"] = filepaths["pr"].replace(" ", "+")
+        filepaths["tasmin"] = filepaths["tasmin"].replace(" ", "+")
+        filepaths["tasmax"] = filepaths["tasmax"].replace(" ", "+")
```

Comment:
Do you mean put the `+` signs back in in the `test` file rather than this file? This issue only happens after calling `client.get`, so it only gets fixed when putting the `+` signs back in this file.

Reply:
Hmm that's interesting.  If that's the case we can leave it as if for now and take care of it when we get to 19.